src/write_gcm.c

```diff
@@ -1135,28 +1135,49 @@ static int wg_oauth2_sign(unsigned char *signature, size_t sig_capacity,
     ERROR("write_gcm: signature buffer not big enough.");
     return -1;
   }
-  EVP_MD_CTX ctx;
-  EVP_SignInit(&ctx, EVP_sha256());
+  
+  #if OPENSSL_VERSION_NUMBER < 0x10100000L
```

Comment:
Let's use the compatibility layer as described in https://wiki.openssl.org/index.php/OpenSSL_1.1.0_Changes .

Reply:
Added the necessary compatibility layer functions so write_gcm works with OpenSSL1.0.